src/simulator/world.py

```diff
@@ -151,7 +152,8 @@ def dump_all_shapes_info(self):
             if self.objects[oid].shape is not None:
                 # TODO: to be replaced with proper handling of compound objects (if cond)
                 shapes_info_dict["shapes"][oid] = self.objects[oid].dump_shape_info()
-
+            else:
+                shapes_info_dict["shapes"][oid] = None
```

Comment:
I suggest you use `defaultdict` to get rid of the `else` part. A `defaultdict`s has a default value for all the unset keys.